src/CoreServices/FSEvents.cs

```diff
@@ -360,6 +417,40 @@ public void Invalidate ()
 				return FSEventStreamGetLatestEventId (handle);
 			}
 		}
+
+		[DllImport (Constants.CoreServicesLibrary)]
+		static extern ulong FSEventStreamGetDeviceBeingWatched (IntPtr handle);
+
+		public ulong GetDevice ()
+		{
+			CheckDisposed ();
+			return FSEventStreamGetDeviceBeingWatched (handle);
+		}
+
+		[Mac (10, 9)]
+		[DllImport (Constants.CoreServicesLibrary)]
+		static extern bool FSEventStreamSetExclusionPaths (IntPtr handle, IntPtr pathsToExclude);
+
+		public bool SetExclusionPaths (NSArray pathsToExclude)
+		{
+			if (pathsToExclude == null) {
+				throw new ArgumentNullException (nameof (pathsToExclude));
+			}
+			CheckDisposed ();
+			return FSEventStreamSetExclusionPaths (handle, pathsToExclude.Handle);
+		}
+
+		[DllImport (Constants.CoreServicesLibrary)]
+		static extern void FSEventStreamSetDispatchQueue (IntPtr handle, DispatchQueue queue);
```

Comment:
Hmmm this looks VERY VERY funny. Is a exter C call that takes https://developer.apple.com/documentation/dispatch/dispatch_queue_t?language=objc as a second parameter. I would expect it to be an IntPtr, so you cannot pass the DispatchQueue object, but its handle.

Reply:
I think that mapping is taken care of by the runtime (https://docs.microsoft.com/en-us/xamarin/cross-platform/macios/binding/objective-c-libraries?tabs=macos)

Reply:
@mandel-macaque good catch :)

@prmallic indeed it's documented... but for **ObjC** bindings. What you're doing are p/invokes, i.e. **C** bindings, which is defined by .net (not our own generator)